core/src/main/java/com/linecorp/armeria/server/Http2ServerConnectionHandler.java

```diff
@@ -118,11 +124,54 @@ private void maybeInitializeKeepAliveHandler(ChannelHandlerContext ctx) {
         }
     }
 
-    private void destroyKeepAliveHandler() {
+    private void cancelScheduledTasks() {
+        gracefulConnectionShutdownHandler.cancel();
         keepAliveHandler.destroy();
     }
 
     KeepAliveHandler keepAliveHandler() {
         return keepAliveHandler;
     }
+
+    @Override
+    public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exception {
+        if (evt instanceof InitiateConnectionShutdown) {
+            setGoAwayDebugMessage("app-requested");
+            gracefulConnectionShutdownHandler.updateDrainDuration(
+                    ((InitiateConnectionShutdown) evt).drainDurationMicros());
+            ctx.channel().close();
```

Comment:
1. Add return after closing a channel?
```suggestion
            gracefulConnectionShutdownHandler.updateDrainDuration(
                    ((InitiateConnectionShutdown) evt).drainDurationMicros());
            ctx.channel().close();
            return;
```
2. I imagine that `InitiateConnectionShutdown` events are simultaneously triggered by a throttling decorator.
   It would be inefficient to call ctx.channel().close() constantly from multiple requests.
   Could we reschedule the ongoing drainFuture using `gracefulConnectionShutdownHandler.updateDrainDuration()`
   and call `ctx.channel().close()` only once?

Reply:
Didn't realize that calling `ctx.channel().close()` multiple times is expensive, fixed.